### PR TITLE
What's Here for Set RDoc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Categorized and linked overview of Set methods.

### CODE REVIEW

File: lib/set.rb [hunk 1]
```diff
@@ -1,5 +1,4 @@
 # frozen_string_literal: true
-# :markup: markdown
```
Comment (BurdetteLamar) [inline lib/set.rb:2]: This usage fouled the definition list items (<tt># #foo:: Returns a foo.</tt>), so I removed it.  This required the other changes in the existing doc, up to line 57.

File: lib/set.rb [hunk 2]
```diff
@@ -38,30 +36,128 @@
...
+#
+# #length, #size:: Returns the count of elements.
+# #empty?:: Returns whether the set has no elements.
+# #include?, #member?, #===:: Returns whether a given object is an element in the set.
+# #subset?, {<=}[#method-i-3C-3D]:: Returns whether a given object is a subset of the set.
+# #proper_subset?, {<}[#method-i-3C]:: Returns whether a given enumerable is a proper subset of the set.
+# #superset?, {<=}[#method-i-3E-3D]:: Returns whether a given enumerable is a superset of the set.
+# #proper_superset?, {>}[#method-i-3E]:: Returns whether a given enumerable is a proper superset of the set.
+# #disjoint?:: Returns +true+ if the set and a given enumerable have no common elements, +false+ otherwise.
+# #intersect?:: Returns +true+ if the set and a given enumerable have any common elements, +false+ othewise.
```
Comment (marcandre) [inline lib/set.rb:115]: Maybe these should be in comparing?

**Review by jeremyevans — COMMENTED** [review]
This pull request should be closed, and a similar one filed upstream in https://github.com/ruby/set.

Changing back from markdown to rdoc needs to be approved by the `set` maintainer (@knu), who converted the documentation from rdoc to markdown about a month ago (maybe previously, but it was only merged into Ruby about a month ago).  That should not be done in the same commit that adds new documentation.  Alternatively, you could just adjust the new documentation to use markdown format.

I can review the upstream pull request to add documentation, after @knu has approved changing the documentation format (or after the new documentation is converted to markdown if not changing the format).

File: lib/set.rb (hunk 2 above)
Comment (BurdetteLamar) [reply lib/set.rb:115]: Each returns boolean, thus query;ing each also does a comparision, thus comparing.  I thought (elsewhere) that some should be in more than one category, but @jeremyevans bridled at that.

Comment (marcandre) [reply lib/set.rb:115]: I distinguish the two by the type of argument they take, not by the return type.
Comparing: `foo(set) => bool`
Querying: `foo(obj) => bool`
In particular, `<=>` returns the same type of information as `<=`, `>=`, ...

**Comment by BurdetteLamar** [comment]
Any way I could have known about ruby/set?

**Comment by jeremyevans** [comment]
> @jeremyevans, any way I could have known about ruby/set?

`doc/maintainers.rdoc` lists all upstreams for standard libraries.

**Comment by BurdetteLamar** [comment]
> > @jeremyevans, any way I could have known about ruby/set?
> 
> `doc/maintainers.rdoc` lists all upstreams for standard libraries.

Thanks, @jeremyevans.

Not sure what to do now.  I can revise as markup and put into new PR at ruby/set.  However, in markup I can't get the definition lists to work when the "term" is a link, or even a non-word.  Also, the auto-links are obviously links, which is good, but the explicit links are not obviously links (no bgcolor or font change), which I think is a great loss.

I'm reluctant to propose a format (re-)change to knu.  I'm sure he has his reasons for the change to markdown.

Also, don't want to give up on this, as I think that the Set doc can benefit greatly from this What's Here.

Any guidance welcome.

File: lib/set.rb [hunk 3]
```diff
@@ -38,30 +36,128 @@
 # * When a string is to be stored, a frozen copy of the string is
 #   stored instead unless the original string is already frozen.
 #
-# ## Comparison
+# == Comparison
 #
 # The comparison operators `<`, `>`, `<=`, and `>=` are implemented as
```
Comment (nobu) [inline lib/set.rb:41]: Backquoting code is markdown, doesn't work in rdoc.

**Comment by BurdetteLamar** [comment]
What @jeremyevans said.

**Pull request closed without merging**